src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py

```diff
@@ -316,7 +316,7 @@ def _generate_examples(self, files, metadata_files, split_name, add_metadata, ad
                                 pa_metadata_table = pa_metadata_table.drop(["file_name"])
                                 metadata_dir = os.path.dirname(metadata_file)
                                 metadata_dict = {
-                                    file_name: sample_metadata
+                                    os.path.normpath(file_name): sample_metadata
```

Comment:
`os.path.normpath` inserts the OS-specific file separator, so we need to call `replace("\\", "/")` to make sure it's "/" on all the platforms to avoid the CI error.

```suggestion
                                    os.path.normpath(file_name).replace("\\", "/"): sample_metadata
```

PS: this also means we should remove the `pa_file_name_array = pc.replace_substring(pa_file_name_array, pattern="\\", replacement="/")` line from the script as it's no longer needed.